ldms/src/ldmsd/ldmsd_request_util.c

```diff
@@ -873,7 +873,7 @@ ldmsd_req_attr_t ldmsd_req_attr_get_by_id(char *request, uint32_t attr_id)
 {
 	ldmsd_req_hdr_t req = (ldmsd_req_hdr_t)request;
 	ldmsd_req_attr_t attr = ldmsd_first_attr(req);
-	while (attr->discrim) {
+	while (attr && attr->discrim == 1) {
```

Comment:
Why do we need to compare  `attr->discrim == 1`? Can we leave it as `attr && attr->discrim`? I didn't go through whether all places have initialized attr->discrim to either 1 or 0. Are you certain?

Reply:
I think I am. I changed it to 1 because all the setter set it to 1.

Reply:
I just checked. Either 0 or 1 is assigned to `attr->discrim`. This is safe for now. The attribute data structure would be gone when we merge the new JSON configuration protocol.